### PR TITLE
Sergkanz/move projects2

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Fix for these bugs:

https://github.com/Microsoft/ApplicationInsights-dotnet/issues/307
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/46
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/355
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/287

### CODE REVIEW

File: Test/TestFramework/Shared/StubTelemetryInitializer.cs [hunk 1]
```diff
@@ -19,7 +20,7 @@ public StubTelemetryInitializer()
         /// <summary>
         /// Gets or sets the callback invoked by the <see cref="Initialize"/> method.
         /// </summary>
-        public TelemetryAction OnInitialize { get; set; }
+        public Action<ITelemetry> OnInitialize = item => { };
```
Comment (dnduffy) [inline Test/TestFramework/Shared/StubTelemetryInitializer.cs]: I'm not familiar with this syntax, is this saying the default getter returns item but there is an empty method for the setter?
Why not OnInitialize { get; } = item;?

File: Test/TestFramework/Shared/StubWebRequest.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
     using System.IO;
     using System.Net;
     using System.Threading.Tasks;
-#if NET45 || NET46
+#if CORE_PCL || NET45 || NET46
```
Comment (dnduffy) [inline Test/TestFramework/Shared/StubWebRequest.cs]: I thought we were getting rid of PCL and moving to .NET Core?

File: src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs [hunk 3]
```diff
@@ -136,7 +136,14 @@ private void DequeueAndSend(TimeSpan timeout)
         /// </summary>
         private Task Send(IEnumerable<ITelemetry> telemetryItems, TimeSpan timeout)
         {
-            if (telemetryItems == null || !telemetryItems.Any())
+            byte[] data = null;
+
+            if (telemetryItems != null)
```
Comment (dnduffy) [inline src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs]: Should this also check for && telemetryItems.Any()?

Comment (SergeyKanzhelev) [inline src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs]: yes, that was the source of multiple enumerations.

File: Test/TestFramework/Shared/StubWebRequest.cs (hunk 2 above)
Comment (SergeyKanzhelev) [inline Test/TestFramework/Shared/StubWebRequest.cs]: Yes. For now just merging test classes to minimize duplicaiton

File: Test/TestFramework/Shared/StubTelemetryInitializer.cs (hunk 1 above)
Comment (SergeyKanzhelev) [inline Test/TestFramework/Shared/StubTelemetryInitializer.cs]: this thing defines the `Action` with the default value of an action that does nothing

Comment (dnduffy) [inline Test/TestFramework/Shared/StubTelemetryInitializer.cs]: Sorry I did not read clearly and see Action<T>.

**Pull request merged**